### DISCUSSION TITLE
Checkbox.Checked doesn't accept a String nor Binding as a value in AXAML.

### DISCUSSION THREAD

**Discussion by theweekday-z** [AvaloniaUI/Avalonia#6544] [Q&A]
Steps to reproduce should be straightforward enough. I'm using the Avalonia Extension for VS19. 

If I create a signature such as:
`public void MyFunction(object sender, RoutedEventArgs eventArgs) => MyArray.Insert(0, sender.ToString());`

And try to implement it with:
`<CheckBox Content="..." Checked="MyFunction"/>`

Then I'll be told that "Checked" does not accept a String type, nor a binding type. It implies that I need to reference an EventHandler, but MyFunction _should_ be the event handler. I am definitely missing something basic here, unsure what. 

The [Routed Events](https://docs.avaloniaui.net/docs/input/routed-events#routed-event-handlers-and-xaml) documentation in Avalonia suggest I'm doing this correctly, but perhaps there is a difference in behavior between the Button Click event, and the CheckBox Checked event.

**Comment by maxkatz6** [comment]
Just for case, where is MyFunction defined? It should be in axaml file code behind (i.e. *.axaml.cs)

**Reply by theweekday-z** [reply]
Yep. Same file as where InitializeComponent, and so on are defined. MainWindow.axaml to MainWindow.axaml.cs iirc.

**Reply by maxkatz6** [reply]
Is should work. Can you provide any project to reproduce?

**Reply by theweekday-z** [reply]
Yep, [sure](https://github.com/wellinthat-case/rti-overlay).

**Reply by maxkatz6** [reply]
Your example works perfectly for me.
I just had to add these handlers to the check box:
```
<CheckBox Content="AoA" Foreground="WhiteSmoke" FontWeight="Light" Checked="varChecked" Unchecked="varUnchecked" />
```

**Reply by maxkatz6** [reply]
> told that "Checked" does not accept a String type, nor a binding type

Just for case, if you edit something outside of opened XAML file, you will need to rebuild the project in order to see changes in the previewer. Or build and run it simply to see changes, there won't be any errors during compilation related to that.
